### ISSUE TITLE
Missing/aliased paths

### ISSUE THREAD

**Issue by strogonoff** [ietf-ribose/bibxml-data-archive-sync#3]
Per https://github.com/ietf-ribose/bibxml-service/issues/110, there are two possible paths:

- http://xml2rfc.tools.ietf.org/public/rfc/bibxml-ids/reference.I-D.draft-sparks-sipcore-multiple-reasons-00.xml
- http://xml2rfc.tools.ietf.org/public/rfc/bibxml-ids/reference.I-D.sparks-sipcore-multiple-reasons-00.xml

For unrelated reason, our system could not resolve one of them (this was fixed since).

The actual problem, though, is that the fallback to xml2rfc archive path also failed, because this archive does not have the file with that name. (Apparently, it doesn’t have files without the `draft-` prefix.)

Could we either have all aliases present (this would imply some data duplication though), or specify aliasing behavior for all datasets where it’s present so it can be replicated in BibXML service xml2rfc fallback logic?

**Comment by ronaldtse** [comment]
From that ticket you quoted:
> (However, it could be a problem that our bibxml data archive repository does not contain these path variants, I wonder if I should file a separate issue about xml2rfc path resolution tool having to support these aliases even if they aren’t listed separately in the archive.)

I think this is to be addressed by software implementation not data duplication. We should not duplicate all these file names that have identical content.

**Comment by strogonoff** [comment]
We should have either the full dataset of all paths (even if duplicated), or an exact description of how the xml2rfc.tools.ietf.org does custom path resolution for all directories.

It looks like it’s more involved than we anticipated and we need to reproduce that behavior if we don’t have all paths statically archived.

**Comment by strogonoff** [comment]
So far my description is this:

- for `…/bibxml3/*` paths, the server will try to resolve a version of the path both with and without `draft-` filename prefix

That’s the resolution scenario that I encountered so far, but once we assume they can exist there can be more of them, and my understanding of the one scenario above can be incomplete as well. Ideally, someone who developed or maintains the xml2rfc web server could shed some light.

**Comment by strogonoff** [comment]
For the record, note that we already have duplication in https://github.com/ietf-ribose/bibxml-data-archive/tree/main/bibxml2 (where each file is presented with and without underscore prefix), so there is a precedent.

**Comment by ronaldtse** [comment]
Agree with @strogonoff on the current situation and the resolutions.

**Comment by strogonoff** [comment]
@ronaldtse

So what I need to proceed is either 1) clarification on xml2rfc aliasing behavior, or 2) adjustment of the rsync endpoint so that it “un-aliases” files during output for `bibxml3` and outputs both prefixed and unprefixed variants (like it’s apparently done for `bibxml2` already).

**Comment by ronaldtse** [comment]
1. has to be done by IETF.
2. For the `draft-{name}` and `{name}` patterns in bibxml3 (which is the Internet-Drafts repo), can't we implement a path redirect for `{name}` if a requested path did not have a prefix of "draft-"? We don't want duplicated data.

**Comment by strogonoff** [comment]
> * For the `draft-{name}` and `{name}` patterns in bibxml3 (which is the Internet-Drafts repo), can't we implement a path redirect for `{name}` if a requested path did not have a prefix of "draft-"? We don't want duplicated data.

It depends on what will be done by IETF—either paths will be unaliased in rsync dataset and duplicated, or aliasing behavior will be clarified. It depends what’s more feasible for IETF.

I am fine with duplicating data because it simplifies fallback logic we need to maintain. Fallback should be as simple and foolproof as possible, it’s a temporary dataset that could be eliminated when all paths are mapped and migrated. Having to duplicate resolution logic is more error-prone and if done wrong it can lead to unresolved xml2rfc paths which won’t be detected by xml2rfc path resolution management tool.

**Comment by ronaldtse** [comment]
@strogonoff my only issue with duplicating data is that I won't remember which entry was duplicated and should be removed. If we keep track of those I don't mind.

**Comment by strogonoff** [comment]
> @strogonoff my only issue with duplicating data is that I won't remember which entry was duplicated and should be removed.

I think there is no need to remove individual paths here, ever. The purpose is for legacy tools to be able to resolve _all of the old_ path variations reliably, and this means if we don’t have a match the fallback path must be present. This means, ideally all paths should be included in fallback dataset (least ambiguous), or we need to port any custom resolution logic from the old xml2rfc web server. And once we have tools switch to the new API and/or 100% manually mapped and resolving paths, then we can disconnect the entire archive (not remove individual paths).

Here I just want to make it clear that making xml2rfc path resolution tool work fully requires knowing exact aliasing behavior _or_ full unaliased data in the dataset, whichever way is easier, which means action from someone who runs the rsync endpoint and/or the preexisting xml2rfc web service.

**Comment by strogonoff** [comment]
Filed https://github.com/ietf-ribose/bibxml-service/issues/126 with a question to Robert.

**Comment by ronaldtse** [comment]
@strogonoff can we close this now?

**Comment by strogonoff** [comment]
Yes.

**Comment by strogonoff** [comment]
No. We have more aliases (https://github.com/ietf-ribose/bibxml-service/issues/35#issuecomment-1046107318)
